src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPeripheral.java

```diff
@@ -0,0 +1,250 @@
+
+    @Override
+    public boolean equals(IPeripheral other)
+    {
+        if (other != null && other instanceof SpeakerPeripheral)
+        {
+            SpeakerPeripheral otherSpeaker = (SpeakerPeripheral) other;
+            return otherSpeaker.m_speaker == m_speaker;
+        }
+
+        else
+        {
+            return false;
+        }
+
+    }
+
+
+    @Override
+    public void attach(@Nonnull IComputerAccess computerAccess)
+    {
+    }
+
+    @Override
+    public void detach(@Nonnull IComputerAccess computerAccess)
+    {
+    }
+
+    @Nonnull
+    @Override
+    public String getType()
+    {
+        return "speaker";
+    }
+
+    @Nonnull
+    @Override
+    public String[] getMethodNames()
+    {
+        return new String[] {
+                "playSound", // Plays sound at resourceLocator
+                "playNote" // Plays note
+        };
+    }
+
+    @Override
+    public Object[] callMethod(@Nonnull IComputerAccess computerAccess, @Nonnull ILuaContext context, int methodIndex, @Nonnull Object[] args) throws LuaException
+    {
+        switch (methodIndex)
+        {
+            // playsound
+            case 0:
+            {
+                return playSound(args, context, false);
+            }
+
+            // playnote
+            case 1:
+            {
+                return playNote(args, context);
+            }
+
+            default:
+            {
+                throw new LuaException("Method index out of range!");
+            }
+
+        }
+    }
+
+    private Object[] playNote(Object[] arguments, ILuaContext context) throws LuaException
+    {
+        double volume = 1f;
+        double pitch = 1f;
+
+        // Check if arguments are correct
+        if (arguments.length == 0) // Too few args
+        {
+            throw new LuaException("Expected string, number (optional), number (optional)");
+        }
+
+        if (!(arguments[0] instanceof String)) // Arg wrong type
+        {
+            throw new LuaException("Expected string, number (optional), number (optional)");
+        }
+
+        if (!SoundEvent.REGISTRY.containsKey(new ResourceLocation("block.note." + arguments[0])))
+        {
+            throw new LuaException("Invalid instrument, \"" + arguments[0] + "\"!");
+        }
+
+        if (arguments.length > 1)
+        {
+            if (!(arguments[1] instanceof Double))   // Arg wrong type
+            {
+                throw new LuaException("Expected string, number (optional), number (optional)");
+            }
+            volume = ((Double) arguments[1]).floatValue();
+
+        }
+
+        if (arguments.length > 2)
+        {
+            if (!(arguments[1] instanceof Double))  // Arg wrong type
+            {
+                throw new LuaException("Expected string, number (optional), number (optional)");
+            }
+            pitch = ((Double) arguments[2]).floatValue();
+        }
+
+        if (arguments.length > 3)
```

Comment:
I would just remove this check. it's perfectly okay in Lua style to pass extra stuff on the end that isn't used